src/linter.js

```diff
@@ -337,16 +337,21 @@ export default class Linter {
       });
   }
 
-  extractMetadata({ _Xpi=Xpi, _console=console,
-                    _Directory=Directory } = {}) {
+  extractMetadata({ _Crx=Crx, _console=console, _Directory=Directory,
+                    _Xpi=Xpi } = {}) {
     return checkMinNodeVersion()
       .then(() => {
         return this.checkFileExists(this.packagePath);
       })
       .then((stats) => {
         if (stats.isFile() === true) {
-          log.info('Package is a file. Attempting to parse as an .xpi/.zip');
-          this.io = new _Xpi(this.packagePath);
+          if (this.packagePath.endsWith('.crx')) {
+            log.info('Package is a file ending in .crx; parsing as a CRX');
+            this.io = new _Crx(this.packagePath);
+          } else {
+            log.info('Package is a file. Attempting to parse as an .xpi/.zip');
```

Comment:
English is super weird, oddly enough it's "an Xpi" the same as "an X-Ray". 🙃

- tofumatt

> On 7 Jun 2016, at 02:36, Christopher Grebs <notifications@github.com> wrote:
> 
> In src/linter.js:
> 
> >      return checkMinNodeVersion()
> >        .then(() => {
> >          return this.checkFileExists(this.packagePath);
> >        })
> >        .then((stats) => {
> >          if (stats.isFile() === true) {
> > -          log.info('Package is a file. Attempting to parse as an .xpi/.zip');
> > -          this.io = new _Xpi(this.packagePath);
> > +          if (this.packagePath.endsWith('.crx')) {
> > +            log.info('Package is a file ending in .crx; parsing as a CRX');
> > +            this.io = new _Crx(this.packagePath);
> > +          } else {
> > +            log.info('Package is a file. Attempting to parse as an .xpi/.zip');
> "parse as a .xpi/.zip" am I wrong?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>